### PR TITLE
Stardew Valley: Properly support Universal Tracker

### PR DESCRIPTION
**Pull request by Jouramie**
## What is this fixing or adding?
This fix 
- Universal Tracker when used with Stardew Valley. This however only work with seeds generated with this fix. 
- The stability of the entrance randomizer, which could generate different layout when playing with mods.

The fix in itself is pretty simple. Instead of using the seed provider by the multiworld, a new seed is generated and saved in the slot data. Then the saved seed is reused to generate with UT. 

Note that there is a limitation to this fix. The generation and the player need the same version of the apworld (which should usually be the case). If someone is outdated, the random might end up in a different state. For instance, you will not be able to track a 5.x.x game if you have 6.x.x installed.

## How was this tested?
Unit test, and also generated a couple seed to confirm it works. Compared spoiler with added log to confirm bundles and entrances are the same.

## If this makes graphical changes, please attach screenshots.
N/A

### CODE REVIEW

**Review by ScipioWright — APPROVED** [review]
Looks fine to me, and from what I can tell, core is fine with how you're doing the seed in slot data.
Note that if a yaml has random for any of its options, it will silently show the accessible locations as based on whatever that internal gen decided to random to, so it may be a good idea to either:
1. Set the player's options based on the contents of re_gen_passthrough
2. Display some sort of warning about the player having random options

...or you can just let it be since it's the user using UT wrong anyway

**Comment by Jouramie** [comment]
I think I've read somewhere that from a generation point of view, it's impossible to know if yaml contained random or an actual value. Also, not all options are added to slot data, so we will still miss some of them. 

As I'm unsure what would be the best way to support random value in yaml, so I'll leave it as it is right now. As you said, users can still replace the random in their yaml by the value that was actually rolled.

**Review by EmilyV99 — APPROVED** [review]
LGTM (didn't test myself)

File: worlds/stardew_valley/test/stability/TestStability.py [hunk 1]
```diff
@@ -54,3 +53,6 @@ def test_all_locations_and_items_are_the_same_between_two_generations(self):
             # We check that the actual rule has the same order to make sure it is evaluated in the same order,
             #  so performance tests are repeatable as much as possible.
             self.assertEqual(rule_a, rule_b, f"Location rule of {location_a} at index {i} is different between both executions. Seed={seed}")
+
+        for key, value in result_a["slot_data"].items():
```
Comment (agilbert1412) [inline worlds/stardew_valley/test/stability/TestStability.py:57]: I would personally put each of these 4 `for`s in a subtest, as I find it more clean in the test results, but I won't die on this hill. Can ignore.

File: worlds/stardew_valley/test/rules/TestBundles.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ class TestRaccoonBundlesLogic(SVTestBase):
         options.BundlePrice: options.BundlePrice.option_normal,
         options.Craftsanity: options.Craftsanity.option_all,
     }
-    seed = 1234  # Magic seed that does what I want. Might need to get changed if we change the randomness behavior of raccoon bundles
+    seed = 2  # Magic seed that does what I want. Might need to get changed if we change the randomness behavior of raccoon bundles
```
Comment (agilbert1412) [inline worlds/stardew_valley/test/rules/TestBundles.py:40]: image

**Review by agilbert1412 — APPROVED** [review]

File: worlds/stardew_valley/test/stability/TestStability.py (hunk 1 above)
Comment (Jouramie) [reply worlds/stardew_valley/test/stability/TestStability.py:57]: I will indeed ignore, thanks tho.

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**